src/platform/ASR/PlatformManagerImpl.cpp

```diff
@@ -299,11 +311,13 @@ void PlatformManagerImpl::SetEventFlags(uint32_t flags)
 
 void PlatformManagerImpl::HandleTimerEvent(void)
 {
+    LockChipStack();
```

Comment:
This is probably not the right locking scope.  Compare the locking scope that the FreeRTOS implementation has....

Reply:
I understand,your purpose is to lock the chip stack when 'Handle Timer and Event', and immediate pop all queue . We changed the '_RunEventLoop' implementation copied from FreeRTOS, that makes the implementation look cleaner. we use lega rtos instead of freertos because that allows for compatibility with more RTOS